### PR TITLE
#152882650 Set up database models

### PR DESCRIPTION
**Pull request by mikey2020**
#### What does this PR do?
Sets up database models

#### Description of Task to be completed?
To set up database models

#### What are the relevant pivotal tracker stories?
#152882650

#### Screenshots (if appropriate)
nil

### CODE REVIEW

File: src/server/models/session.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    challengeId: DataTypes.INTEGER
+  });
+
+  Session.associate = (models) => {
+    Session.belongsTo(models.Challenge, {
+      foreignKey: 'challengeId',
+      onDelete: 'CASACADE'
+    });
+
+    Session.belongsToMany(models.User, {
```
Comment (tomipaul) [inline src/server/models/session.js]: @mikey2020 
Between Session and User, there is a one to many relationship;
A user can have many sessions but a session can only belong to a user

File: src/server/models/Challenge.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = (sequelize, DataTypes) => {
+  const Challenge = sequelize.define('Challenge', {
+    name: DataTypes.STRING,
+    duration: DataTypes.INTEGER
+  });
+
+  Challenge.associate = (models) => {
```
Comment (tomipaul) [inline src/server/models/Challenge.js]: According to the drawn schema, I think the challenge model is meant to have moduleId.
The current implementation puts challengeId on the module model which might not be appropriate considering it is a **one-to-many** relationship

Comment (tomipaul) [reply src/server/models/Challenge.js]: Perhaps you can look into changing this to `belongsTo` so we can have appropriate accessor methods

File: src/server/models/module.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module.exports = (sequelize, DataTypes) => {
+  const Module = sequelize.define('Module', {
+    name: DataTypes.STRING,
+    totalDuration: DataTypes.INTEGER
+  });
+
+  Module.associate = (models) => {
+    Module.hasMany(models.Challenge, {
```
Comment (tomipaul) [inline src/server/models/module.js:8]: A `moduleId` foreign key on the challenge model; I think this satisfies our schema requirements

File: src/server/models/session.js (hunk 1 above)
Comment (tomipaul) [reply src/server/models/session.js]: The current implementation is a **many-to-many** relationship
Session should have a `userId` foreign key

File: src/server/models/user.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    email: {
+      type: DataTypes.STRING,
+      unique: true
+    },
+    password: DataTypes.STRING
+  });
+
+  User.associate = (models) => {
+    User.belongsToMany(models.Session, {
```
Comment (tomipaul) [inline src/server/models/user.js]: This also follows the above discussion about `User -> Sessions`

File: package.json [hunk 5]
```diff
@@ -6,9 +6,12 @@
   "scripts": {
     "start": "node dist/server/app.js",
     "start:dev": "babel-watch src/server/app.js",
+    "test": "NODE_ENV=test npm run db:migrate && mocha src/server/test/**  --require babel-polyfill --compilers js:babel-register --exit && npm run undo:migrate",
```
Comment (tomipaul) [inline package.json:9]: @mikey2020 @ayodelevm @3m3kalionel what do you feel about us using **jest** for testing both on client side and server-side?

**Comment by mikey2020** [comment]
All changes requested has been made

File: package.json (hunk 5 above)
Comment (ayodelevm) [reply package.json:9]: Make sense

**Pull request merged**